src/Compilers/Core/Portable/PEWriter/Units.cs

```diff
@@ -77,9 +77,10 @@ internal interface IModule : IUnit, IModuleReference
         bool GenerateVisualBasicStylePdb { get; }
 
         /// <summary>
-        /// Public types defined in other modules making up this assembly and to which other assemblies may refer to via this assembly.
+        /// Public types defined in other modules making up this assembly and to which other assemblies may refer to via this assembly
+        /// followed by types forwarded to another assembly.
         /// </summary>
-        IEnumerable<ITypeReference> GetExportedTypes(EmitContext context);
+        ImmutableArray<ExportedType> GetExportedTypes(DiagnosticBag diagnostics);
```

Comment:
Yes, passing stuff to methods that they don't need avoids accidental unnecessary dependencies.